tutorials/yes-no-game/full-game/yesnogame.edn

```diff
@@ -58,11 +59,53 @@
   0 >  .seconds-elapsed
   0 > .total-score)
 
+(defwire fisher-yates-shuffle
+  = .input-sequence
+
+  (Setup
+   (Sequence .output-sequence)
+   0 >= .input-sequence-length
+   0 >= .x)
+
+  (Clear .output-sequence)
+  (Count .input-sequence) > .input-sequence-length
+  0 > .x
+
+  (Repeat
+   :Action
+   (->
+    (Count .input-sequence) > .input-sequence-length
+    (RandomInt :Max .input-sequence-length) > .x
+    .input-sequence (Take .x) (Push .output-sequence)
+    (Erase .x .input-sequence))
+   :Forever true
+   :Until (-> (Count .input-sequence) (Is 0)))
+
+  .output-sequence)
```

Comment:
It is a working implementation but it is not efficient: instead of needing two sequences and erasing one, the shuffle can be done *in-place*:

```suggestion
(defwire fisher-yates-shuffle
  (ExpectSeq) = .sequence
  (Count .sequence) (Math.Subtract 1) >= .n

  (Repeat
   (-> (RandomInt :Max .n) >= .i
       .sequence
       (| (Take .i) >= .j)
       (| (Take .n) >= .l)

       [.i .l] (Assoc .sequence)
       [.n .j] (Assoc .sequence)

       (Math.Dec .n))
   :Forever true
   :Until (-> .n (Is 0)))
  
  .sequence)
```